ring/token_range.go

```diff
@@ -0,0 +1,136 @@
+package ring
+
+import (
+	"math"
+
+	"github.com/pkg/errors"
+	"golang.org/x/exp/slices" // using exp/slices until moving to go 1.21.
+)
+
+// TokenRanges describes token ranges owned by an instance.
+// It consists of [start, end] pairs, where both start and end are inclusive.
+type TokenRanges []uint32
```

Comment:
I would be more expressive here and I would say something about the length of the slice, and also about the special cases you handle in the implementation.
E.g., this slice contains an even number of elements, elements at even and odd positions represent range starts and ends respectively, etc.

Reply:
Documenting implementation details would commit to this implementation. Given the early nature of the development, we may still want or need to change it. I prefer to hide implementation details in supporting functions/methods, and only document what's necessary for clients to use.

Reply:
I agree that we shouldn’t document implementationsl details, but giving an example saying something like “_e.g., [5, 10, 20, 30] means that the corresponding instance covers tokens 5-10 and 20-30_” wouldn’t damage.